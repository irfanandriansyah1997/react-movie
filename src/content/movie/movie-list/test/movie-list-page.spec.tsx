import React from 'react';
import { mount } from 'enzyme';

import { HashRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import MovieListPage from '../movie-list.page';
import StoreBuilder from '../../../../shared/modules/redux/builder/store.builder';

describe('Testing Snapshot Pages', () => {
    it('should render card movie and search bar', (done: () => void) => {
        const store = StoreBuilder.singleton();

        const pages = mount(
            <HashRouter>
                <Provider store={store}>
                    <MovieListPage />
                </Provider>
            </HashRouter>
        );

        setTimeout(() => {
            expect(pages.html()).toMatchSnapshot();
            done();
        }, 3000) 

    });
});