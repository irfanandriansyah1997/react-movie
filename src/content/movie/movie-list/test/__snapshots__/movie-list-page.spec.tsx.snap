// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot Pages should render card movie and search bar 1`] = `"<div class=\\"firstRow\\"><div class=\\"ui-container\\"></div></div><div class=\\"secondRow\\"><div class=\\"ui-container\\"><div class=\\"input\\"><label for=\\"search\\"><span class=\\"material-icons\\">search</span></label><input id=\\"search\\" type=\\"text\\" autocomplete=\\"off\\" placeholder=\\"Ketikan Film Yang Akan Anda Cari\\" value=\\"Batman\\"><button type=\\"button\\">Cari Film</button></div></div></div><div class=\\"ui-container\\"><p class=\\"text default result\\" style=\\"font-weight: 400; text-align: center;\\">Hasil pencarian<b>\\"Batman\\"</b>sebanyak 391 data</p><div class=\\"list\\"><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman Begins\\" src=\\"https://m.media-amazon.com/images/M/MV5BOTY4YjI2N2MtYmFlMC00ZjcyLTg3YjEtMDQyM2ZjYzQ5YWFkXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt0372784\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman Begins</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">2005</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman v Superman: Dawn of Justice\\" src=\\"https://m.media-amazon.com/images/M/MV5BYThjYzcyYzItNTVjNy00NDk0LTgwMWQtYjMwNmNlNWJhMzMyXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt2975590\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman v Superman: Dawn of Justice</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">2016</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman\\" src=\\"https://m.media-amazon.com/images/M/MV5BMTYwNjAyODIyMF5BMl5BanBnXkFtZTYwNDMwMDk2._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt0096895\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">1989</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman Returns\\" src=\\"https://m.media-amazon.com/images/M/MV5BOGZmYzVkMmItM2NiOS00MDI3LWI4ZWQtMTg0YWZkODRkMmViXkEyXkFqcGdeQXVyODY0NzcxNw@@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt0103776\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman Returns</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">1992</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman Forever\\" src=\\"https://m.media-amazon.com/images/M/MV5BNDdjYmFiYWEtYzBhZS00YTZkLWFlODgtY2I5MDE0NzZmMDljXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt0112462\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman Forever</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">1995</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman &amp; Robin\\" src=\\"https://m.media-amazon.com/images/M/MV5BMGQ5YTM1NmMtYmIxYy00N2VmLWJhZTYtN2EwYTY3MWFhOTczXkEyXkFqcGdeQXVyNTA2NTI0MTY@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt0118688\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman &amp; Robin</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">1997</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"The Lego Batman Movie\\" src=\\"https://m.media-amazon.com/images/M/MV5BMTcyNTEyOTY0M15BMl5BanBnXkFtZTgwOTAyNzU3MDI@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt4116284\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">The Lego Batman Movie</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">2017</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman: The Animated Series\\" src=\\"https://m.media-amazon.com/images/M/MV5BOTM3MTRkZjQtYjBkMy00YWE1LTkxOTQtNDQyNGY0YjYzNzAzXkEyXkFqcGdeQXVyOTgwMzk1MTA@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt0103359\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman: The Animated Series</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">series</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">1992â€“1995</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman: Under the Red Hood\\" src=\\"https://m.media-amazon.com/images/M/MV5BNmY4ZDZjY2UtOWFiYy00MjhjLThmMjctOTQ2NjYxZGRjYmNlL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt1569923\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman: Under the Red Hood</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">2010</p></li></ul></a></div></div><div class=\\"card ui-movies-card card\\"><div class=\\"container\\"><img alt=\\"Batman: The Dark Knight Returns, Part 1\\" src=\\"https://m.media-amazon.com/images/M/MV5BMzIxMDkxNDM2M15BMl5BanBnXkFtZTcwMDA5ODY1OQ@@._V1_SX300.jpg\\" role=\\"presentation\\" class=\\"image ui-movies-card__image\\"><a class=\\"content\\" href=\\"#/detail/tt2313197\\"><div class=\\"title\\"><p class=\\"text heading ui-movies-card__title\\" style=\\"color: rgb(34, 43, 69); font-weight: 500;\\">Batman: The Dark Knight Returns, Part 1</p></div><ul class=\\"list\\"><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">movie</p></li><li><p class=\\"text caption listItem ui-movies-card__list-item\\" style=\\"color: rgb(105, 118, 132); font-weight: 400;\\">2012</p></li></ul></a></div></div></div></div><div class=\\"dialog\\"><div><span role=\\"presentation\\" class=\\"material-icons\\">close</span><img src=\\"\\" alt=\\"Detail Movie\\"></div></div>"`;
